toolchain/diagnostics/coverage_test.cpp

```diff
@@ -41,10 +42,9 @@ constexpr DiagnosticKind UntestedDiagnosticKinds[] = {
 
 // Looks for diagnostic kinds that aren't covered by a file_test.
 TEST(Coverage, DiagnosticKind) {
-  Testing::TestKindCoverage(absl::GetFlag(FLAGS_testdata_manifest),
-                            R"(^ *// CHECK:STDERR: .*\.carbon:.* \[(\w+)\]$)",
-                            llvm::ArrayRef(DiagnosticKinds),
-                            llvm::ArrayRef(UntestedDiagnosticKinds));
+  Testing::TestKindCoverage(
+      absl::GetFlag(FLAGS_testdata_manifest), R"( \[([A-Z]\w*)\]$)",
```

Comment:
This regex will match most anything now, is it still effectively testing something?

Reply:
I agree it's quite wider, but seems to work fine for now.
I can think of alternatives but we might want to adjust it when we have a new use case that doesn't fit.
Let me know what you think.

Reply:
I'm similarly concerned that this is more broad than it needs to be. Does this work if you change it back to only looking in diagnostic output, e.g.:

```suggestion
      absl::GetFlag(FLAGS_testdata_manifest), R"(^ *// CHECK:STDERR: .* \[(\w+)\]$)",
```

Note, in general I think it's good to figure out ways to make regexes restrictive. The check as it stands would match something like an array in source code, so a test like:

```
var x: [N]
```

Reply:
Makes sense, thanks, done.